dashboard/src/components/ChartList/ChartList.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { shallow } from "enzyme";
+import * as React from "react";
+
+import { IChart, IChartState } from "../../shared/types";
+import { CardGrid } from "../Card";
+import { NotFoundErrorAlert } from "../ErrorAlert";
+import PageHeader from "../PageHeader";
+import SearchFilter from "../SearchFilter";
+import ChartList from "./ChartList";
+
+const defaultChartState = {
+  isFetching: false,
+  selected: {} as IChartState["selected"],
+  items: [],
+} as IChartState;
+const defaultProps = {
+  charts: defaultChartState,
+  repo: "stable",
+  filter: "",
+  fetchCharts: jest.fn(),
+  pushSearchFilter: jest.fn(),
+};
+
+it("propagates the filter from the props", () => {
+  const wrapper = shallow(<ChartList {...defaultProps} filter="foo" />);
+  expect(wrapper.state().filter).toBe("foo");
+});
+
+describe("renderization", () => {
+  it("should render an error if no charts are found", () => {
+    const wrapper = shallow(<ChartList {...defaultProps} />);
+    expect(wrapper.find(NotFoundErrorAlert).exists()).toBe(true);
+    expect(wrapper.find(".ChartList").exists()).toBe(false);
+    expect(
+      wrapper
+        .find(NotFoundErrorAlert)
+        .children()
+        .text(),
+    ).toContain("Manage your Helm chart repositories");
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it("should render a loading message if is still fetching charts", () => {
```

Comment:
this is probably going to conflict with @migmartri's PR #644

Reply:
Not yet. That patch does not include this view.

I am doing another PR for this component so I'll skip the tests and rebase once @andresmgot landes this.